corehq/apps/data_interfaces/models.py

```diff
@@ -340,18 +347,26 @@ def criteria_match(self, case, now):
                 (case.server_modified_on > (now - timedelta(days=self.server_modified_boundary))):
             return False
 
-        for criteria in self.memoized_criteria:
-            try:
-                result = criteria.definition.matches(case, now)
-            except CaseNotFound:
-                # This might happen if the criteria references a parent case and the
-                # parent case is not found
-                result = False
-
-            if not result:
-                return False
-
-        return True
+        if self.criteria_operator == 'ANY':
+            for criteria in self.memoized_criteria:
+                try:
+                    if criteria.definition.matches(case, now):
+                        return True
+                except CaseNotFound:
+                    continue
+            return False
+        else:
+            for criteria in self.memoized_criteria:
+                try:
+                    result = criteria.definition.matches(case, now)
+                except CaseNotFound:
+                    # This might happen if the criteria references a parent case and the
+                    # parent case is not found
+                    result = False
+
+                if not result:
+                    return False
+            return True
```

Comment:
I think this could be dried up a bit:

```python
def _evaluate_criteria(criteria):
    try:
        return criteria.definition.matches(case, now)
    except CaseNotFound:
        # This might happen if the criteria references a parent case and the
        # parent case is not found
        result = False

results = (_evaluate_criteria(criteria) for criteria in self.memoized_criteria)
if self.criteria_operator == 'ANY':
    return any(results)
else:
    return all(results)
```